internal/sql/repository/chartConfig/ChartRepository.go

```diff
@@ -73,6 +118,151 @@ type ChartRepositoryImpl struct {
+	for i, appNameInclude := range appNameIncludes {
+		if i == 0 {
+			appNameIncludesQuery += fmt.Sprintf("app_name LIKE '%s' ", appNameInclude)
+		} else {
+			appNameIncludesQuery += fmt.Sprintf("OR app_name LIKE '%s'", appNameInclude)
+		}
+	}
+	var appNameExcludesQuery string
+	for i, appNameExclude := range appNameExcludes {
+		if i == 0 {
+			appNameExcludesQuery += fmt.Sprintf("app_name NOT LIKE '%s' ", appNameExclude)
+		} else {
+			appNameExcludesQuery += fmt.Sprintf("AND app_name NOT LIKE '%s'", appNameExclude)
+		}
+	}
+	appNameQuery := fmt.Sprintf("%s AND %s", appNameIncludesQuery, appNameExcludesQuery)
+	err := repositoryImpl.dbConnection.
+		Model(&apps).Join("INNER JOIN charts ON app.id = app_id").
+		Join("INNER JOIN chart_env_config_override ON charts.id = chart_id").
+		Where(appNameQuery, " AND target_environment = ? AND chart_env_config_override.latest = ?", envId, true).
+		Select()
+	return apps, err
+}
+func (repositoryImpl ChartRepositoryImpl) FindBulkChartsByAppNameSubstring(appNameIncludes []string, appNameExcludes []string) ([]*Chart, error) {
+	var charts []*Chart
+	//Concatenating string according to sql LIKE operator required format
+	var appNameIncludesQuery string
+	for i, appNameInclude := range appNameIncludes {
+		if i == 0 {
+			appNameIncludesQuery += fmt.Sprintf("app_name LIKE '%s' ", appNameInclude)
+		} else {
+			appNameIncludesQuery += fmt.Sprintf("OR app_name LIKE '%s'", appNameInclude)
+		}
+	}
+	var appNameExcludesQuery string
+	for i, appNameExclude := range appNameExcludes {
+		if i == 0 {
+			appNameExcludesQuery += fmt.Sprintf("app_name NOT LIKE '%s' ", appNameExclude)
+		} else {
+			appNameExcludesQuery += fmt.Sprintf("AND app_name NOT LIKE '%s'", appNameExclude)
+		}
+	}
+	appNameQuery := fmt.Sprintf("%s AND %s", appNameIncludesQuery, appNameExcludesQuery)
+	err := repositoryImpl.dbConnection.
+		Model(&charts).Join("INNER JOIN app ON app.id=app_id ").
+		Where(appNameQuery, " AND latest = ?", true).
+		Select()
+	return charts, err
+}
+func (repositoryImpl ChartRepositoryImpl) FindBulkChartsEnvByAppNameSubstring(appNameIncludes []string, appNameExcludes []string, envId int) ([]*ChartEnvConfigOverride, error) {
+	var charts []*ChartEnvConfigOverride
+	//Concatenating string according to sql LIKE operator required format
+	var appNameIncludesQuery string
+	for i, appNameInclude := range appNameIncludes {
+		if i == 0 {
+			appNameIncludesQuery += fmt.Sprintf("app_name LIKE '%s' ", appNameInclude)
+		} else {
+			appNameIncludesQuery += fmt.Sprintf("OR app_name LIKE '%s'", appNameInclude)
+		}
+	}
+	var appNameExcludesQuery string
+	for i, appNameExclude := range appNameExcludes {
+		if i == 0 {
+			appNameExcludesQuery += fmt.Sprintf("app_name NOT LIKE '%s' ", appNameExclude)
+		} else {
+			appNameExcludesQuery += fmt.Sprintf("AND app_name NOT LIKE '%s'", appNameExclude)
+		}
+	}
+	appNameQuery := fmt.Sprintf("%s AND %s", appNameIncludesQuery, appNameExcludesQuery)
```

Comment:
what if there len(appNameExcludesQuery) == 0, this will be invalid query

Reply:
added check for this